app/uk/gov/hmrc/organisationsmatchingapi/controllers/BaseApiController.scala

```diff
@@ -51,6 +56,71 @@ abstract class BaseApiController (cc: ControllerComponents) extends BackendContr
           "message" -> "The request body does not conform to the schema.",
           "errors" -> Json.toJson(errors.toList))))
 
+  private[controllers] def recoveryWithAudit(correlationId: Option[String], matchId: String, url: String)(
+    implicit request: RequestHeader,
+    auditHelper: AuditHelper): PartialFunction[Throwable, Result] = {
+    case _: MatchNotFoundException => {
+      auditHelper.auditApiFailure(correlationId, matchId, request, url, "Not Found")
```

Comment:
I don't think this is an ApiFailure - a MatchNotFound is still a successful response with no match.